src/main/java/com/enterprise/eventmanagementsystem/controller/EventController.java

```diff
@@ -167,27 +166,13 @@ public String delete(@PathVariable(value = "id") int id) throws Exception {
 
     /**
      * Search for an event that exists.
-     * @param str
+     * @param searchText
      * @return all events that were similar to the search term entered into the search field.
      * @throws Exception
      */
     @GetMapping("/search/{str}")
     @ResponseBody
-    public List<Event> searchEvents(@PathVariable(value = "str") String str) throws Exception {
-        List<Event> eventList =  new LinkedList<>();
-        for (Event event:eventService.fetchAll()) {
-            if(event.getName().contains(str) || event.getDescription().contains(str)){
-                eventList.add(event);
-            }
-        }
-        return eventList;
+    public List<Event> searchEvents(@PathVariable(value = "str") String searchText) throws Exception{
+        return eventService.searchEvents(searchText);
     }
```

Comment:
In this controller method, you have some logic that would also be best suited in the business logic / service layer. As you can see here, I deleted the logic in your method, and moved it to a service method.

Reply:
Good idea.